Code/Framework/AzToolsFramework/Tests/EditorViewportHelperTests.cpp

```diff
@@ -0,0 +1,247 @@
+#include <AzToolsFramework/UnitTest/Mocks/MockFocusModeInterface.h>
+#include <AzToolsFramework/ViewportSelection/EditorHelpers.h>
+
+namespace UnitTest
+{
+    class EditorViewportOnlyDrawHelpersForSelectedEntityFixture
+        : public ToolsApplicationFixture
+        , public AzFramework::EntityDebugDisplayEventBus::MultiHandler
+    {
+    public:
+        inline static constexpr AzFramework::ViewportId TestViewportId = 2468;
+
+        static bool isEntitySelected(AZ::EntityId entityId)
+        {
+            AzToolsFramework::EntityIdList entityIds;
+            AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+                entityIds, &AzToolsFramework::ToolsApplicationRequests::GetSelectedEntities);
+
+            if (AZStd::ranges::find(entityIds, entityId) != entityIds.end())
+            {
+                return true;
+            }
+            return false;
+        }
+
+        void SetUpEditorFixtureImpl() override
+        {
+            // Setup entity to use for EntityDebugDisplayEventBus and tests
+            AZ::Entity* entity1 = nullptr;
+            m_entityId = CreateDefaultEditorEntity("DebugHelpersEntity", &entity1);
+
+            AZ::Entity* entity2 = nullptr;
+            m_entityId2 = CreateDefaultEditorEntity("DebugHelpersEntity2", &entity2);
+
+            AzFramework::EntityDebugDisplayEventBus::MultiHandler::BusConnect(entity2->GetId());
+            AzFramework::EntityDebugDisplayEventBus::MultiHandler::BusConnect(entity1->GetId());
+
+            // DebugDisplay to use in DisplayHelpers
+            AzFramework::DebugDisplayRequestBus::BusPtr debugDisplayBus;
+            AzFramework::DebugDisplayRequestBus::Bind(debugDisplayBus, TestViewportId);
+            m_debugDisplay = AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
+
+            m_cameraState = AzFramework::CreateDefaultCamera(AZ::Transform::CreateIdentity(), AzFramework::ScreenSize(1024, 768));
+
+            m_entityVisibleEntityDataCacheMock = AZStd::make_unique<::testing::NiceMock<MockEditorVisibleEntityDataCacheInterface>>();
+            m_editorHelpers = AZStd::make_unique<AzToolsFramework::EditorHelpers>(m_entityVisibleEntityDataCacheMock.get());
+            m_viewportSettings = AZStd::make_unique<ViewportSettingsTestImpl>();
+
+            m_viewportSettings->Connect(TestViewportId);
+            m_viewportSettings->m_helpersVisible = true;
+            m_viewportSettings->m_iconsVisible = true;
+
+            using ::testing::_;
+            using ::testing::Return;
+
+            EXPECT_CALL(*m_entityVisibleEntityDataCacheMock, GetVisibleEntityId(_))
+                .WillOnce(Return(m_entityId))
+                .WillOnce(Return(m_entityId2));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, VisibleEntityDataCount()).WillByDefault(Return(2));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, IsVisibleEntityVisible(_)).WillByDefault(Return(true));
```

Comment:
question: Do we definitely want the expectation mock setup in the fixture? Do we always want these return values? It just seems like maybe this should be decided per test? I could be wrong though just want to double check

Reply:
It's set like this because the entityVisibleEntityDataCache are the entities visible in the viewport, we want those two entities visible at all times but the helpers to show up based on the selection if the setting is enabled.

Reply:
Okay understood thanks! :+1: